toogoodtogo_ha_mqtt_bridge/main.py

```diff
@@ -199,25 +201,71 @@ def check_for_removed_stores(shops: []):
 def loop(event):
     logger.info("Starting loop")
 
+    create_data_dir()
     token_exits = check_existing_token_file()
     tgtg_client.login()
     if not token_exits and tgtg_client.access_token:
         write_token_file()
 
+    event.wait(calc_next_run())
     while True:
         logger.debug("Loop run started")
         if not check():
             logger.error("Loop was not successfully.")
         else:
             logger.debug("Loop run finished")
+            watchdog.timeout = calc_timeout()
             watchdog.reset()
-        event.wait(settings.tgtg.every_n_minutes * 60)
 
+        event.wait(calc_next_run())
 
-def watchdog_handler():
-    logger.error(f"Watchdog handler fired! No pull in the last {settings.tgtg.every_n_minutes} minutes!")
 
-    os._exit(1)  # easy way to die from within a thread
+def calc_next_run():
+    tgtg = settings.get("tgtg")
+
+    if "polling_schedule" not in tgtg:
+        logger.error("No polling_schedule found in settings")
+        os._exit(1)
```

Comment:
This code (log and exit) is found multiple times.
Maybe create some `exit_from_thread(message: str, return_code: int = 1)` methode for that? This way this kinda dirty usage of `os._exit()` can be easily changed some day, e.g. when the threading part might be changed to something better.

Maybe `logger.exception` might also be nice at thay point? https://docs.python.org/3/library/logging.html#logging.Logger.exception

Reply:
I will change this, thanks!

Reply:
Fix with [0c8de9e](https://github.com/MaxWinterstein/toogoodtogo-ha-mqtt-bridge/pull/28/commits/0c8de9ecc3e97f0ba5600092ab2b54e61e8909a3)